span-normalizer/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/jaeger/JaegerSpanPreProcessor.java

```diff
@@ -94,6 +97,9 @@ PreProcessedSpan preProcessSpan(Span span) {
     String tenantId = maybeTenantId.get();
 
     if (spanFilter.shouldDropSpan(span, tags)) {
+      if (DROPPED_SPANS_RATE_LIMITER.tryAcquire()) {
+        LOG.info("Dropped span: {}", span);
```

Comment:
Would be nice to also log the reason for dropping but that's a little bit more involving.

Reply:
Please check again.